content/en/blog/_posts/protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,283 @@
+
+There can also be other scenarios where you want to control the order of scheduling or order of eviction of pods.
+
+## PriorityClasses in Kubernetes
+
+PriorityClass is a cluster-wide object/resource in a Kubernetes cluster and is a part of the scheduling.k8s.io/v1 API group. It contains a mapping of the PriorityClass name (which can be defined in priorityclass.metadata.name) and an integer value (which can be defined in priorityclass.value). This is the value that the pods get and is used for defining the priority.
+
+Do you think Kubernetes itself uses PriorityClasses? The answer is yes. When we create a cluster by kubeadm or even in a managed Kubernetes service (for example, in AKS-Azure Kubernetes Service), PriorityClasses are widely used to safeguard the pods which are hosted on the master node, i.e., all the control-plane components/pods and other pods as well, like CoreDNS, metrics-server, kube-proxy, etc.
+
+PriorityClass makes sure that these pods are up and alive to serve the requests and make sure that the cluster is up and running.
+
+```console
+$ kubectl get pc
+NAME                      VALUE        GLOBAL-DEFAULT   AGE
+system-cluster-critical   2000000000   false            82m
+system-node-critical      2000001000   false            82m
+```
+
+_Note: pc is the short name of PriorityClass._
+
+The below diagram shows exactly how it works and also contains the details of the example, which will come in the upcoming section (PriorityClass HandsOn Example).
+
+![Whenever there is a new incoming pod, this is how kube-scheduler decides their priority and puts them in a queue](decision-tree.svg)
+
+### Pod priority and preemption
+
+Pod preemption is a Kubernetes feature that allows the cluster to evict/preempt lower-priority pods over high-priority pods in the event of a resource crunch. To learn more about this concept, please refer to the Kubernetes [official documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/).
+
+Pod priority and preemption are Kubernetes features:
+- Which help in deciding a priority while scheduling pods.
+- Which check if the mission-critical pods are up and running and are not evicted.
+- Decide how the cluster will behave in the event of a resource crunch.
+- Decide which lower priority pods are preempted (evicted) from the nodes when there are resource crunch and high priority pods are available in the queue for scheduling.
+
+### PriorityClass requirements
+
+1. Decision on which all PriorityClasses are needed.
+   Example: Based on environments, or based on the kind of pods, or based on the kind of applications.
+2. A globalDefault PriorityClass resource in the cluster (otherwise the pods without priorityClassName will be treated as 0).
```

Comment:
```suggestion
2. Which PriorityClass resource will be the default for the cluster (otherwise the pods without a `priorityClassName` will be treated as priority 0).
```